PWGJE/EMCALJetTasks/macros/AddTaskJetResponseMaker.C

```diff
@@ -101,8 +101,11 @@ AliJetResponseMaker* AddTaskJetResponseMaker(
   jetCont2->SetFlavourCut(jetTagging);
   jetCont2->SetMaxTrackPt(1000); // disable default 100 GeV/c track cut for particle level jets
 
-    
+#ifdef __CLING__
+  jetTask->SetMatching(static_cast<AliJetResponseMaker::MatchingType>(matching), maxDistance1, maxDistance2);
+#else
   jetTask->SetMatching(matching, maxDistance1, maxDistance2);
+#endif
```

Comment:
Why is this necessary? That cling requires the correct type is clear, but the static_cast should in principle do no harm under cint.

Reply:
I actually don't know the reason for doing this - I was inclined to just use the static_cast, but I noticed that you used this approach in a number of places, and I assumed that you had a reason for it. [For example, your commit here](https://github.com/alisw/AliPhysics/commit/363df1459f567e784a188bcc25616177af4cb9cf).  Do you recall what the CINT problem was?

In any case, I'm happy to remove it if it's not needed - just let me know what you prefer.

Reply:
just as a side comment : you can declare this function in a namespace along the .h/.cpp of AliJetResponseMaker code and convert your .C macro to a return MY_NS::AddTaskJetResponseMaker (my_signature ). This way will not matter if it is root 5 or 6 as it is compiled code (and for a usage point of view is equivalent as the AddTask macros are not modified locally, they are used anyway from the $ALICE software locations)

Reply:
Thanks for the suggestion - that is actually what I usually do for my own tasks, but I didn't in this case as I don't really own this task.

@aiola Is it okay with you if I just convert this to a compiled AddTask?

And while we're on the topic - would anyone object to moving the sample task to a compiled AddTask?

Reply:
I agree with @adriansev proposal. It has been done now for a good fraction of new classes and I think if we come across we might fix the old ones as well.

Reply:
Okay, I will this close this PR, move some tasks to compiled AddTasks, and then make a new PR.